bip-0360.mediawiki

```diff
@@ -1,705 +1,401 @@
+  </li>
 
-<source>
-stack = [pubkey||signature]
-tapscript = [OP_PUSHDATA HASH256(expected_pubkey), OP_CHECKMLSIG]
-</source>
+  <li>
+    Facilitate gradual integration of quantum-resistant features that can be carried out iteratively as quantum computers evolve. This approach encourages responsiveness to the current threat-level, while avoiding heavy-handedness in our reactions to a potential threat.
+    <p>We designed P2TSH with an eye towards integrating post-quantum signatures in the future, without proposing more complex changes while CRQCs are still in their infancy.</p>
+  </li>
+</ol>
 
-1. OP_PUSHDATA HASH256(expected_pubkey) updates the stack to [HASH256(expected_pubkey), pubkey||signature]
-2. OP_CHECKMLSIG pops HASH256(expected_pubkey) and pubkey||signature, checks HASH256(expected_pubkey) == pubkey and verifies signature against pubkey.
+=== P2TSH Trade-Offs ===
 
-==== Future considerations ====
+While P2TR outputs (and ECC-based key-path spends) will remain an option for folks wishing to use them – we aim to be clear about the tradeoffs of using P2TSH outputs, which disable the key-path spend for the benefit of quantum protection.
 
-Additional follow-on BIPs will be needed to implement PQ signature algorithms, signature aggregation, and full BIP 32 compatibility
-(if possible) <ref name="bip-32">BIP 32 relies on elliptic curve operations to derive keys from xpubs to support
-watch-only wallets, which PQC schemes may not support.</ref>. However, until specialized quantum cryptography hardware
-is widespread and signature aggregation schemes are thoroughly vetted, P2QRH addresses are an intermediate solution 
-to quantum threats.
+First off, P2TSH script-path spends are larger than P2TR key-path spends, because you have to include the control byte and post the script with every transaction. That said, script-path spends from P2TSH outputs will be slightly smaller than script-path spends from P2TR outputs because there is no need to reveal any internal public key. For a more complete comparison of input witness sizes, the “Transaction Size and Fees” section may be reviewed later in this proposal.
 
-== Specification ==
+Additionally, there is a privacy tradeoff when comparing P2TSH and P2TR, which is that users will functionally reveal that they are spending to a script tree whenever they are using P2TSH output, since P2TSH outputs can only be spent via script-path spend. In P2TR when you spend an output as a key-path spend, you don’t reveal any script or if you had any scripts at all. This trade-off only exists when comparing P2TR key-path spends to P2TSH script-path spends; P2TR and P2TSH provide the same level of privacy when both are script-path spends.
```

Comment:
```suggestion
Additionally, there is a privacy tradeoff when comparing P2TSH and P2TR, which is that users will functionally reveal that they are spending from a script tree whenever they are using P2TSH output, since P2TSH outputs can only be spent via the script-path spend. In P2TR when you spend an output as a key-path spend, you don’t reveal any script or if you had any scripts at all. This trade-off only exists when comparing P2TR key-path spends to P2TSH script-path spends; P2TR and P2TSH provide the same level of privacy when both are script-path spends.
```